worker/state/worker.go

```diff
@@ -166,31 +166,8 @@ func (w *modelStateWorker) loop() error {
 		_, _ = w.pool.Remove(w.modelUUID)
 	}()
 
-	// Jitter the interval so that each model doesn't attempt to connect to
-	// mongo at potentially the same time.
-	interval := w.pingInterval + jitter(time.Millisecond*200)
-
-	// If the state ping fails, attempt to retry the ping, before returning.
-	var pingErr error
-	for attempt := 0; attempt < 5; {
-		select {
-		case <-w.tomb.Dying():
-			return tomb.ErrDying
-		case <-time.After(interval):
-			if pingErr = st.Ping(); pingErr != nil {
-				// Reduce the next ping interval to fail early in case mongo
-				// has actually died. This should prevent the worst case
-				// scenario of a large initial ping interval.
-				interval = maxDuration(interval/2, time.Second)
-				attempt++
-				continue
-			}
-			interval = w.pingInterval
-			attempt = 0
-		}
-	}
-
-	return errors.Annotate(pingErr, "state ping failed")
+	err = st.Ping()
+	return errors.Annotate(err, "state ping failed")
```

Comment:
This exits the loop, closing the model's state.

What you need is restoration of the original inner loop, which is:
```go
for {
		select {
		case <-w.tomb.Dying():
			return tomb.ErrDying
		case <-time.After(w.pingInterval):
			if err := st.Ping(); err != nil {
				return errors.Annotate(err, "state ping failed")
			}
		}
	}
```

Reply:
Oh I see sorry! So this worker continuously pings, failing if the ping fails?